File: 단계별로 풀어보기/재귀/10870/피보나치 수 5.cpp
#include <iostream>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int first = 0;
	int second = 1;
	int result, n;

	cin >> n;

	if (n == 0) {
		result = first;
	}
	else if(n==1){
		result = second;
	}
	else {
		for (int i = 2; i < n + 1; i++) {
			result = first + second;
			first = second;
			second = result;
		}
	}

	cout << result;
	return 0;
}

File: 단계별로 풀어보기/재귀/10870/피보나치 수 5.py
import sys

def init():
    n = int(sys.stdin.readline())

    first = 0
    second = 1
    
    if n == 0:
        result = 0
    elif n == 1:
        result = 1
    else:
        for i in range(2, n+1):
            result = first + second
            first = second
            second = result
            
    print(result)
    
init()
